plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,270 @@
+
+            columnChunks.forEach(partialColumns -> getStatsFutures.add(CompletableFuture.supplyAsync(() -> {
+                final List<String> columnsNames = partialColumns.stream()
+                        .map(Column::getName)
+                        .collect(toImmutableList());
+
+                final GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(partition.getDatabaseName())
+                        .withTableName(partition.getTableName())
+                        .withColumnNames(columnsNames)
+                        .withPartitionValues(partition.getValues());
+
+                return glueClient.getColumnStatisticsForPartition(request);
+            }, readExecutor)));
+
+            final HiveBasicStatistics tableStatistics = getHiveBasicStatistics(partition.getParameters());
+            final ImmutableMap.Builder<String, HiveColumnStatistics> columnStatsMapBuilder = ImmutableMap.builder();
+            for (CompletableFuture<GetColumnStatisticsForPartitionResult> future : getStatsFutures.build()) {
+                final GetColumnStatisticsForPartitionResult partitionColumnStats = MoreFutures.getFutureValue(future, TrinoException.class);
+                partitionColumnStats.getColumnStatisticsList()
+                        .forEach(col -> columnStatsMapBuilder.put(col.getColumnName(),
+                                GlueStatConverter.fromGlueColumnStatistics(col.getStatisticsData(), tableStatistics.getRowCount())));
+            }
+
+            return columnStatsMapBuilder.build();
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStatistics> columnStatistics)
+    {
+        try {
+            final HiveBasicStatistics tableStats = getHiveBasicStatistics(table.getParameters());
+
+            if (columnStatistics.isEmpty()) {
+                final List<CompletableFuture<Void>> writeFutures = Stream.concat(table.getDataColumns().stream(), table.getPartitionColumns().stream())
+                        .map(column -> CompletableFuture.runAsync(() -> glueClient.deleteColumnStatisticsForTable(
+                                new DeleteColumnStatisticsForTableRequest()
+                                        .withCatalogId(catalogId)
+                                        .withDatabaseName(table.getDatabaseName())
+                                        .withTableName(table.getTableName())
+                                        .withColumnName(column.getName())), writeExecutor))
+                        .collect(Collectors.toUnmodifiableList());
+
+                for (CompletableFuture<Void> writeFuture : writeFutures) {
+                    MoreFutures.getFutureValue(writeFuture);
+                }
```

Comment:
maybe
```
                getFutureValue(allOf(writeFutures.toArray(CompletableFuture[]::new)));
```

I am not super convinced. I would be more if there was `allOf` which takes a list ....